app/src/main/java/com/google/droidjump/LevelsFragment.java

```diff
@@ -16,26 +16,70 @@
 
 package com.google.droidjump;
 
-import static androidx.navigation.Navigation.findNavController;
+import android.annotation.SuppressLint;
+import android.content.Context;
 import android.os.Bundle;
+import android.support.annotation.Nullable;
 import android.support.v4.app.Fragment;
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
-import android.widget.Button;
+import android.widget.GridView;
+import android.widget.ImageButton;
+import androidx.navigation.Navigation;
+import java.util.Objects;
 
 /**
  * Displays Levels Screen.
  */
 public class LevelsFragment extends Fragment {
 
+    private MainActivity activity;
+    private int[] levels;
+    private LevelsAdapter adapter;
+
     @Override
-    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
-        View rootView = inflater.inflate(R.layout.levels_screen, container, /* attachToRoot= */ false);
-        Button levelPlayButton = rootView.findViewById(R.id.level_play);
-        levelPlayButton.setOnClickListener(view -> {
-            findNavController(view).navigate(R.id.action_levels_screen_to_game_screen);
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        activity = Objects.requireNonNull((MainActivity) getActivity());
+        levels = createLevels(activity.getLevelsCount());
+        Context context = getContext();
+        adapter = new LevelsAdapter(Objects.requireNonNull(context), levels,
+                activity.getCurrentLevel());
+    }
+
+    @SuppressLint("SetTextI18n")
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        View rootView = inflater.inflate(R.layout.levels_screen, container,
+                /* attachToRoot= */ false);
+
+        // Finding gridView and putting levels to it.
+        GridView gridView = rootView.findViewById(R.id.levels_grid_view);
+        gridView.setAdapter(Objects.requireNonNull(adapter));
+
+        // Adding onClick events.
+        gridView.setOnItemClickListener((adapterView, view, i, l) -> {
```

Comment:
Please don't use "l" as a variable name as it is almost indistinguishable with "1" in this text style. Give it a meaningful name.

I see you don't even use it - skip it is possible. If not, call it "_" (underscore symbol) or "ignored" or something like that so it can be understood that this variable is not required.

Reply:
Make sense. Fixed.

Reply:
I still see it.